### PR TITLE
MAX! improve stability & actual updates

### PR DESCRIPTION
**Pull request by marcelrv**
* Fixes initialisation issues for MAX! devices during when loading the
bridge
* Improved communication module
* Code simplification
* Experimental: Implementation of actuals update (issue #203)
* Foundation for #201
* Refactoring binding structure

Signed-off-by: Marcel Verpaalen <marcel@verpaalen.com>

### CODE REVIEW

File: addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxDevicesHandler.java [hunk 1]
```diff
@@ -206,6 +249,77 @@ public void onDeviceStateChanged(ThingUID bridge, Device device) {
...
+						device.getSerialNumber(), getThing().getUID());
+
+				originalSetTemp = device.getTemperatureSetpoint();
+				originalMode = device.getMode();
+
+				if (originalMode == ThermostatModeType.MANUAL || originalMode == ThermostatModeType.AUTOMATIC) {
+					BigDecimal temporaryTemp = ((DecimalType) originalSetTemp).toBigDecimal().add(
+							BigDecimal.valueOf(0.5));
+					logger.debug("Actuals Refresh: Setting Temp {}", temporaryTemp);
+					handleCommand(new ChannelUID(getThing().getUID(), CHANNEL_SETTEMP), (Command) new DecimalType(
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxDevicesHandler.java]: do you really mean handleCommand here? This method is supposed to actively send a new temperature setpoint to the device, but here you seem to only want to refresh the locally stored value?

File: addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxDevicesHandler.java [hunk 2]
```diff
@@ -206,6 +249,77 @@ public void onDeviceStateChanged(ThingUID bridge, Device device) {
...
+
+	/**
+	 * Send the commands to restore the original settings for mode & temperature to end the automatic update cycle 
+	 */
+	private synchronized void refreshActualsRestore() {
+		try {
+			refreshingActuals = false;
+			if (originalMode == ThermostatModeType.AUTOMATIC || originalMode == ThermostatModeType.MANUAL) {
+				logger.debug("Finished Actuals Refresh: Restoring Temp {}", originalSetTemp);
+				handleCommand(new ChannelUID(getThing().getUID(), CHANNEL_SETTEMP), (Command) originalSetTemp);
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxDevicesHandler.java]: same here

**Comment by kaikreuzer** [comment]
lgtm, just note my remarks about handleCommand.

File: addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxDevicesHandler.java (hunk 1 above)
Comment (marcelrv) [inline addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxDevicesHandler.java]: Hi @kaikreuzer ,

indeed that's intended here. The MAX! devices only update the actual temperature after the valve position is changed. The only way to change the valve position is to really change the set temperature.

Hence the logic for this new feature is to change the set temp +0.5 C and after 2 minutes change it back to it's original setting.

Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxDevicesHandler.java]: Ah, alright, thanks for the explanation!

**Pull request merged**